show_client/ndp_cli.go

```diff
@@ -0,0 +1,418 @@
+	})
+
+	table.TotalEntries = len(table.Entries)
+	return table
+}
+
+func getInterfaceOidMap() (map[string]string, error) {
+	portQueries := [][]string{
+		{"COUNTERS_DB", "COUNTERS_PORT_NAME_MAP"},
+	}
+	lagQueries := [][]string{
+		{"COUNTERS_DB", "COUNTERS_LAG_NAME_MAP"},
+	}
+
+	portMap, err := GetMapFromQueries(portQueries)
+	if err != nil {
+		return nil, err
+	}
+	lagMap, err := GetMapFromQueries(lagQueries)
+	if err != nil {
+		return nil, err
+	}
+
+	// SONiC interface regex patterns
+	ethRe := regexp.MustCompile(`^Ethernet(\d+)$`)
+	lagRe := regexp.MustCompile(`^PortChannel(\d+)$`)
+	vlanRe := regexp.MustCompile(`^Vlan(\d+)$`)
+	mgmtRe := regexp.MustCompile(`^eth(\d+)$`)
+
+	oidPrefix := len("oid:0x")
+	ifOidMap := make(map[string]string)
+
+	// helper closure to check valid names
+	isValidIfName := func(name string) bool {
+		return ethRe.MatchString(name) ||
+			lagRe.MatchString(name) ||
+			vlanRe.MatchString(name) ||
+			mgmtRe.MatchString(name)
+	}
+
+	for portName, oidVal := range portMap {
+		oidStr, ok := oidVal.(string)
+		if !ok {
+			continue
+		}
+		if isValidIfName(portName) {
+			ifOidMap[oidStr[oidPrefix:]] = portName
+		}
+	}
+	for lagName, oidVal := range lagMap {
+		oidStr, ok := oidVal.(string)
+		if !ok {
+			continue
+		}
+		if isValidIfName(lagName) {
+			ifOidMap[oidStr[oidPrefix:]] = lagName
+		}
+	}
+
+	return ifOidMap, nil
+}
+
+func buildBvidToVlanMap() (map[string]string, error) {
+	queries := [][]string{
+		{"ASIC_DB", "ASIC_STATE:SAI_OBJECT_TYPE_VLAN:*"},
+	}
+
+	vlanData, err := GetMapFromQueries(queries)
+	if err != nil {
+		return nil, err
+	}
+
+	const prefix = "SAI_OBJECT_TYPE_VLAN:"
+	result := make(map[string]string)
+
+	for key, val := range vlanData {
+		if !strings.HasPrefix(key, prefix) {
+			continue
+		}
+
+		bvid := strings.TrimPrefix(key, prefix) // "oid:..."
+
+		ent, ok := val.(map[string]interface{})
+		if !ok {
+			log.Warningf("Unexpected format for VLAN entry %s: %#v", key, val)
+			continue
+		}
+
+		if vlanIDRaw, ok := ent["SAI_VLAN_ATTR_VLAN_ID"]; ok {
+			if vlanIDStr, ok := vlanIDRaw.(string); ok {
+				result[bvid] = vlanIDStr
+			}
+		}
+	}
+
+	return result, nil
+}
+
+func getVlanIDFromBvid(bvid string, bvidMap map[string]string) (string, error) {
+	if vlanID, ok := bvidMap[bvid]; ok {
+		return vlanID, nil
+	}
+	return "", fmt.Errorf("BVID %s not found in VLAN map", bvid)
+}
+
+func getBridgePortMap() (map[string]string, error) {
+	queries := [][]string{
+		{"ASIC_DB", "ASIC_STATE:SAI_OBJECT_TYPE_BRIDGE_PORT:*"},
+	}
+	brPortStr, err := GetMapFromQueries(queries)
+	if err != nil {
+		return nil, err
+	}
+	log.V(6).Infof("SAI_OBJECT_TYPE_BRIDGE_PORT data from query: %v", brPortStr)
+
+	ifBrOidMap := make(map[string]string)
+	oidPrefix := len("oid:0x")
+
+	for key, val := range brPortStr {
+		parts := strings.SplitN(key, ":", 2)
+		if len(parts) < 2 {
+			continue
+		}
+		bridgePortOid := parts[1][oidPrefix:] // strip "oid:0x"
```

Comment:
In this case wouldnt parts[1] be `SAI_OBJECT_TYPE_BRIDGE_PORT:oid:0x3a00000000097b` so [oidPrefix:] wont work as expected?

Reply:
updated with key example and add range check